Add tests for CommandLineEdit

nicoddemus opened the pull request.

nicoddemus committed:
Add tests for CommandLineEdit

hackebrot committed:
Remove blank line at end of file to fix flake8

hackebrot commented:
Fixed the flake8 issue, hope you don't mind.

I personally prefer to not use a class in this particular case, but it's totally up to you!

Looks good to me :+1:

The-Compiler commented:
Sorry for the long delay, I was a bit busy with [the new release](https://github.com/The-Compiler/qutebrowser/releases/tag/v0.2.0) :smile:

Is there a good reason this is using unittest.mock instead of pytest-mock? I thought we were trying to get rid of unittest.mock? :wink:

nicoddemus commented:
@hackebrot thanks for fixing flake8, my bad I missed it.

About the class, I usually prefer free functions too, but in this case I figure we will add more tests for other widgets from `miscwidgets`, and classes are a good way to group those tests. :smile:

nicoddemus committed:
Play nice with other plugins in conftest.py

Some plugins might create their own Item subclasses without
a `fixturenames` attribute. Discovered while taking pytest-flakes
for a spin.

nicoddemus committed:
Use pytest-mock to install QApplication.clipboard mock

nicoddemus commented:
@The-Compiler yeah, not sure why I used `with` at that point... changed to use `pytest-mock`. :smile:

Cheers, :beers:

The-Compiler commented on tests/misc/test_miscwidgets.py:
```diff
@@ -0,0 +1,97 @@
+        cleaning up afterwards.
+        """
+        command_history = lineparser.LimitLineParser('', '', limit=None)
+        objreg.register('command-history', command_history)
```
I don't think `command-history` is needed here - it's only used for `qutebrowser/mainwindow/statusbar/command.py:Command`, but not in `qutebrowser/misc/miscwidgets.py:CommandLineEdit` which is what we're testing here.

nicoddemus committed:
Remove LimitLineParser from test

As suggested by @The-Compiler, this is not really necessary

nicoddemus commented:
@The-Compiler you're right, not sure why I put that in the first place...

Cheers :beers:

The-Compiler committed:
Fix lint.
